tfe/data_source_organization.go

```diff
@@ -0,0 +1,168 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEOrganization() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceTFEOrganizationRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"external_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"collaborator_auth_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"cost_estimation_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"email": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"enterprise_plan": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"owners_team_saml_role_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"permissions": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"can_create_team": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_create_workspace": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_create_workspace_migration": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_destroy": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_traverse": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_update": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_update_api_token": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_update_oauth": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"can_update_sentinel": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"saml_enalbed": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"session_remember": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"session_timeout": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"two_factor_confrmant": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceTFEOrganizationRead(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	name := d.Get("name").(string)
+	log.Printf("[DEBUG] Read configuration for Organization: %s", name)
+	org, err := tfeClient.Organizations.Read(ctx, name)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			return fmt.Errorf("Could not read organization '%s'", name)
+		}
+		return fmt.Errorf("Error retrieving organization: %v", err)
+	}
+
+	log.Printf("[DEBUG] Setting Organization Attributes")
+	d.Set("name", org.Name)
+	d.SetId(org.Name)
```

Comment:
I used the organization name as the main identifier, even though there is an external_id, because we currently use the `name` as the main identifier.